src/builder/QuarkBuilderBase.sol

```diff
@@ -407,25 +417,36 @@ contract QuarkBuilderBase {
         uint256 chainId,
         address account,
         uint256 blockTimestamp,
-        bool useQuotecall
+        bool useQuotecall,
+        bool useWrapUpTo
     ) internal pure {
         // Check if inserting wrapOrUnwrap action is necessary
         uint256 assetBalanceOnChain =
             Accounts.getBalanceOnChain(assetSymbol, chainId, chainAccountsList) + supplementalBalance;
         if (assetBalanceOnChain < amount && TokenWrapper.hasWrapperContract(chainId, assetSymbol)) {
-            // If the asset has a wrapper counterpart, wrap/unwrap the token to cover the transferIntent amount
+            // If the asset has a wrapper counterpart, wrap/unwrap the token to cover the amount needed for the intent
             string memory counterpartSymbol = TokenWrapper.getWrapperCounterpartSymbol(chainId, assetSymbol);
 
             // Wrap/unwrap the token to cover the amount
+            uint256 amountToWrap;
```

Comment:
If we move to "all things wrapped maybe" then it might be nice that this logic simplifies a bit.

Reply:
Agreed, having to adjust `amountToWrap` is kind of confusing. I'm probably going to bite the bullet and just change the script to `maybe wrap all`

Reply:
Created the `wrap/unwrap all` variants for ETH and stETH. QuarkBuilder now always wraps/unwraps all.